### ISSUE TITLE
ImportError: cannot import name 'quote' from 'pytube.compat'

### ISSUE THREAD

**Issue by ChristopherKlix** [pytube/pytube#782]
Using pytube v9.6.0 w/ Python 3.8.5
Running in virtual env

### Python code
```py
from pytube import YouTube # <-- ImportError

url = 'https://www.youtube.com/watch?v=KtFFmDGIsa4&ab_channel=EVNautilus'
youtube = YouTube(url)
print(youtube.streams.all())
```
### Commandline output
```bash
$ python run.py
...
File "/path_to_python_script/env/lib/python3.8/site-packages/pytube/extract.py", line 7, in <module>
    from pytube.compat import quote
ImportError: cannot import name 'quote' from 'pytube.compat'
```

**Comment by tfdahlin** [comment]
Hi @ChristopherKlix, this issue is because the pypi package for pytube hasn't been updated in quite a while, and I believe the most recent push to pypi had a problem with it. In order to get pytube to work, please make sure you've uninstalled any version of pytube that you have (I recommend running `pip uninstall pytube pytube3 pytubex`, just to make sure you don't have conflicting packages), and then run `python -m pip install git+https://github.com/nficano/pytube`. This should fix your problem.

**Comment by aryanpanpalia** [comment]
I was having the exact same issue. I did exactly what you said to do in your message and that error is now replaced with another one. 

Traceback (most recent call last):
  File "C:/Users/Yash_K/PycharmProjects/SheetsFromSynth/main.py", line 4, in <module>
    video = YouTube('http://youtube.com/watch?v=9bZkp7q19f0')
  File "C:\Users\Yash_K\PycharmProjects\SheetsFromSynth\venv\lib\site-packages\pytube\__main__.py", line 103, in __init__
    self.descramble()
  File "C:\Users\Yash_K\PycharmProjects\SheetsFromSynth\venv\lib\site-packages\pytube\__main__.py", line 123, in descramble
    self.player_config_args = get_ytplayer_config(self.watch_html)[
  File "C:\Users\Yash_K\PycharmProjects\SheetsFromSynth\venv\lib\site-packages\pytube\extract.py", line 206, in get_ytplayer_config
    caller="get_ytplayer_config", pattern="config_patterns"
pytube.exceptions.RegexMatchError: get_ytplayer_config: could not find match for config_patterns

Do you know how I could fix this? Thanks!

**Comment by tfdahlin** [comment]
hi @aryanpanpalia I can't seem to reproduce the error that you are running into locally. Can you provide more details about your installation? It looks like you're using a virtual environment based on that traceback, are you sure that you've correctly installed the latest version in your virtual environment?

**Comment by ChristopherKlix** [comment]
> Hi @ChristopherKlix, this issue is because the pypi package for pytube hasn't been updated in quite a while, and I believe the most recent push to pypi had a problem with it. In order to get pytube to work, please make sure you've uninstalled any version of pytube that you have (I recommend running `pip uninstall pytube pytube3 pytubex`, just to make sure you don't have conflicting packages), and then run `python -m pip install git+https://github.com/nficano/pytube`. This should fix your problem.

That did solve the issue, thank you very much! Would be great if that would be mentioned in the README.md on the GitHub repo. Or updated to be the official pip package.
But thanks again!

**Comment by tfdahlin** [comment]
glad to hear it @ChristopherKlix -- the pip package actually *just* got updated, and hopefully will be kept more up-to-date going forward, installing from the source was just a temporary workaround in the meantime

**Comment by pmcknig1** [comment]
I am also having this issue.  I installed using pip so pypi is still using an old package.  I confirmed the package pip installed is still using `pytube.compat`, and `pytube.extract` is trying to import `urlencode` and `quote` from it.  I'm sure the fix you listed above will fix the problem but it might nice to move this issue back to open status until pypi reflects these changes so people can find the solution to this issue easier.

**Comment by pmcknig1** [comment]
I am also having this issue.  I installed using pip so pypi is still using an old package.  I confirmed the package pip installed is still using `pytube.compat`, and `pytube.extract` is trying to import `urlencode` and `quote` from it.  I'm sure the fix you listed above will fix the problem but it might nice to move this issue back to open status until pypi reflects these changes so people can find the solution to this issue easier. 

The version installed from pypi was `9.7.0`

**Comment by tfdahlin** [comment]
@pmcknig1 that shouldn't be happening. Did you make sure to uninstall all existing versions of pytube (with `pip uninstall pytube pytube3 pytubex`) before installing? Sometimes the shared namespace from the different forks of pytube has caused weird file conflicts for people.

If this doesn't work, can you provide the full error output from the import, as well as the exact output from installing with pip? It could be an issue with where the library is getting installed, and how your PATH is configured. For example, if you have two installations, it could be finding an old installation first when trying to import the library

**Comment by pmcknig1** [comment]
OK, here is what I "think" happened.  When I initially installed pytube my default python3 version was 3.5.2 in which that version of pip installed the broken pytube 9.7.0. When I tracked the initial error down (Something to do with `__annotations__`) to a 3.6 dependency I installed python 3.8.6.  Pip for 3.8.6 didn't work because the install was broken (I didn't know this yet) so I uninstalled pytube and used pip from python 3.5.2 with the `-t` flag to install into 3.8.6's `site-packages` directory.  So I got the same broken version of pytube.

Once I built 3.8.6 from source and got a working 3.8.6 version of pip, I uninstalled everything from pytube and reinstalled using the correct pip/python version.  Now it appears I have the correct version of pytube even though it still showing
```python
>>> pytube.__version__
'9.7.0'
```
I'm no longer getting the import errors and the source source seems matches whats in github.

Sorry for the mix-up
